doc/rst/technotes/libraries.rst

```diff
@@ -111,23 +205,145 @@ example looks like this:
+
+Initializing and Using Your Library in Python
+---------------------------------------------
+
+Once your ``PYTHONPATH`` is set up and the Python module created, you can
+``import`` the module like a normal Python module.
+
+Similarly to using your library with C, you will need to call a set up function
+to ensure the Chapel runtime and standard modules are initialized, as well as
+a clean up function.
+
+Unlike the C case, the set up function is called ``chpl_setup()`` and will also
+handle initializing your module.   This function will still need to be called
+prior to any Chapel library function calls.
+
+Also unlike the C case, the clean up function is called ``chpl_cleanup()``.
+This function will still need to be called after all the Chapel library function
+calls are finished.
+
+For example:
+
+.. code-block:: Python
+
+   import foo
+
+   foo.chpl_setup()
+
+   foo.baz(7) // Call into a library function
+
+   foo.chpl_cleanup()
+
+.. note::
+
+   Currently the ``chpl_cleanup()`` function will also cause the Python program
+   to exit.  This may change in the future, but for now please ensure your
+   Python functionality is also complete before calling this function.
+
+In Case Something Goes Wrong
+----------------------------
+
+This compilation strategy uses Cython under the covers, generating a
+``chpl_foo.pxd`` file, a ``foo.pyx`` file, and a ``foo.py`` file by default for
+a ``libfoo.a`` / ``libfoo.so``, which are then called using a Cython command
+(this command is rather long due to the need to include the Chapel runtime and
+third-party libraries).  These files are currently left in the same location as
+the generated library - if compilation fails due to generating one or more of
+these files incorrectly, you may be able to modify the file and re-run the
+Cython command yourself.
+
+Please report such errors to us using the strategies defined in
+:ref:`readme-bugs`, regardless!
+
+Using Your Library in Chapel
+============================
+
 Chapel library files cannot be used from Chapel code.  The library files must
 include the chapel runtime and standard modules for use in a non-Chapel program
 and when the library is linked to a Chapel program this leads to multiple
 definitions of these functions.
 
-As mentioned above, this feature is not very sturdy.  Please refer to
+Other Caveats
+=============
+
+Multiple Chapel Libraries
+-------------------------
+
+Multiple Chapel libraries cannot currently be used in the same C or Python
+program.  Each library file must include the chapel runtime and standard modules
+for its own functionality and when two or more libraries are linked to a program
+this leads to multiple definitions of these functions.
+
+
+.. _Exporting Symbols:
+
+Exporting Symbols
+-----------------
+
+At this time, only functions can get exported.  We hope to extend this support
+to types and global variables in the future.
+
+Issues?
```

Comment:
This is technically its own header instead of a subheader of Caveats.  I figured it would be good to have this section as close to the top-level as possible, so it is easier to find.  I probably should make the other place link to it instead of duplicating information, though.  Thanks!